deploy/buildtools/bitbuilder.py

```diff
@@ -0,0 +1,320 @@
+            hash = local("git rev-parse HEAD", capture=True)
+        tag_fsimcommit = hash + is_dirty_str
+
+        assert len(tag_fsimcommit) <= 255, "ERR: aws does not support tags longer than 256 chars for fsimcommit"
+
+        # construct the serialized description from these tags.
+        description = firesim_tags_to_description(tag_buildtriplet, tag_deploytriplet, tag_fsimcommit)
+
+        # if we're unlucky, multiple vivado builds may launch at the same time. so we
+        # append the build node IP + a random string to diff them in s3
+        global_append = "-" + str(env.host_string) + "-" + ''.join(random.SystemRandom().choice(string.ascii_uppercase + string.digits) for _ in range(10)) + ".tar"
+
+        with lcd(f"{local_results_dir}/cl_{tag_buildtriplet}/build/checkpoints/to_aws/"), StreamLogger('stdout'), StreamLogger('stderr'):
+            files = local('ls *.tar', capture=True)
+            rootLogger.debug(files)
+            rootLogger.debug(files.stderr)
+            tarfile = files.split()[-1]
+            s3_tarfile = tarfile + global_append
+            localcap = local('aws s3 cp ' + tarfile + ' s3://' + s3bucket + '/dcp/' + s3_tarfile, capture=True)
+            rootLogger.debug(localcap)
+            rootLogger.debug(localcap.stderr)
+            agfi_afi_ids = local(f"""aws ec2 create-fpga-image --input-storage-location Bucket={s3bucket},Key={"dcp/" + s3_tarfile} --logs-storage-location Bucket={s3bucket},Key={"logs/"} --name "{afiname}" --description "{description}" """, capture=True)
+            rootLogger.debug(agfi_afi_ids)
+            rootLogger.debug(agfi_afi_ids.stderr)
+            rootLogger.debug("create-fpge-image result: " + str(agfi_afi_ids))
+            ids_as_dict = json.loads(agfi_afi_ids)
+            agfi = ids_as_dict["FpgaImageGlobalId"]
+            afi = ids_as_dict["FpgaImageId"]
+            rootLogger.info("Resulting AGFI: " + str(agfi))
+            rootLogger.info("Resulting AFI: " + str(afi))
+
+        rootLogger.info("Waiting for create-fpga-image completion.")
+        checkstate = "pending"
+        with lcd(local_results_dir), StreamLogger('stdout'), StreamLogger('stderr'):
+            while checkstate == "pending":
+                imagestate = local(f"aws ec2 describe-fpga-images --fpga-image-id {afi} | tee AGFI_INFO", capture=True)
+                state_as_dict = json.loads(imagestate)
+                checkstate = state_as_dict["FpgaImages"][0]["State"]["Code"]
+                rootLogger.info("Current state: " + str(checkstate))
+                time.sleep(10)
+
+
+        if checkstate == "available":
+            # copy the image to all regions for the current user
+            copy_afi_to_all_regions(afi)
+
+            message_title = "FireSim FPGA Build Completed"
+            agfi_entry = afiname + ":\n"
+            agfi_entry += "    agfi: " + agfi + "\n"
```

Comment:
I think in this PR we should somehow create a section that changes based on the platform it builds for. I.e change the HWDB to be something like:

```
<NAME>:
  deploy_triplet_override: null
  custom_runtime_config: null
  platform_specific:
      type: aws-fpga
      args:
           agfi: ....
```

Reply:
For now, the PR would just support the AWS version of whatever the API would be but this would make the Vitis integration much easier.

Reply:
Im fine with punting this to a later PR (since we also need to add in the URI changes)